### PR TITLE
Added auth using passport

### PR DESCRIPTION
**Pull request by kaiiyer**
Features Added:
- Server with Node.js and Express
- Setting up Passport
- Encrypted user passwords using bcrypt
- Restricting access to routes for logged in users(Once logged in you don't have to revisit the login page before you actually logout)
- Flash messages
- <b>No DB integrated !!</b>

### CODE REVIEW

**Review by jagannathBhat — CHANGES_REQUESTED** [review]
Can the following changes be made?

1. Delete all .ejs files. We can use the pages I designed instead.
2. Comment the routes leading to the .ejs file in server.js. I will add the routes myself.
3. Make all auth routes starting with '/api'. Like the post request '/register' should be '/api/register'.
4. Do we really need the requirements.txt file. We can install from package.json right?
5. Move all files from hhk-auth to the root folder.

**Review by jagannathBhat — APPROVED** [review]

File: server.js [hunk 1]
```diff
@@ -47,6 +49,7 @@ app.post('/login', checkNotAuthenticated, passport.authenticate('local', {
 	failureFlash: true
 }))
 
+// route to register file
 app.get('/register', checkNotAuthenticated, (req, res) => {
 	res.render('register.ejs')
```
Comment (athul) [inline server.js:54]: Still `.ejs` ?

Comment (kaiiyer) [reply server.js:54]: File deleted aa. Newfile add cheiyumbo rename cheital mati

**Review by athul — APPROVED** [review]
LGTM

**Comment by athul** [comment]
@jagannathbhat Merge when you're ready

**Comment by jagannathBhat** [comment]
done

**Pull request merged**